code/ModelAdmin.php

```diff
@@ -238,6 +252,69 @@ public function getExportFields()
         return singleton($this->modelClass)->summaryFields();
     }
 
+    /**
+     * @deprecated 5.0
+     * @return \SilverStripe\ORM\Search\SearchContext
+     */
+    public function getSearchContext()
+    {
+        Deprecation::notice('5.0', 'Will be removed in favor of GridFieldFilterHeader in 5.0');
+
+        $gridField = $this->getEditForm()->fields()
```

Comment:
Not sure about all the `getEditForm` calls. Sounds kind of heavy just so we can get a single field.

Reply:
Good enough.